pkg/trait/jvm.go

```diff
@@ -169,6 +171,59 @@ func (t *jvmTrait) Apply(e *Environment) error {
 		args = append(args, t.Options...)
 	}
 
+	// Translate HTTP proxy environment variables, that are set by the environment trait,
+	// into corresponding JVM system properties.
+	if HTTPProxy := envvar.Get(container.Env, "HTTP_PROXY"); HTTPProxy != nil {
+		u, err := url.Parse(HTTPProxy.Value)
+		if err != nil {
+			return err
+		}
+		if !util.StringSliceContainsAnyOf(t.Options, "http.proxyHost") {
+			args = append(args, fmt.Sprintf("-Dhttp.proxyHost=%q", u.Hostname()))
+		}
+		if port := u.Port(); !util.StringSliceContainsAnyOf(t.Options, "http.proxyPort") && port != "" {
+			args = append(args, fmt.Sprintf("-Dhttp.proxyPort=%q", u.Port()))
+		}
+		if user := u.User; !util.StringSliceContainsAnyOf(t.Options, "http.proxyUser") && user != nil {
+			args = append(args, fmt.Sprintf("-Dhttp.proxyUser=%q", user.Username()))
+			if password, ok := user.Password(); !util.StringSliceContainsAnyOf(t.Options, "http.proxyUser") && ok {
+				args = append(args, fmt.Sprintf("-Dhttp.proxyPassword=%q", password))
+			}
+		}
+	}
+
+	if HTTPSProxy := envvar.Get(container.Env, "HTTPS_PROXY"); HTTPSProxy != nil {
```

Comment:
Nit: considering that it's the same behavior for HTTP, we may try to create a function, so, any maintenance would be simpler

Reply:
Sure, let's do this in a subsequent PR so as to have this one in the upcoming release.